tests/cabal.project

```diff
@@ -0,0 +1,9 @@
+packages: bytestring-tests.cabal
+          ../bytestring.cabal
```

Comment:
Why include the `bytestring` package here? Doesn't this cause test dependencies to be recompiled after changes to bytestring code?

Reply:
IIRC the test frameworks and text need to use the same version of ByteString as being tested. We compiling the tests with code that also uses Text, which uses ByteString, ... that cyclic thing.  But here we're build the test module, and `bytestring` is just one of the dependencies to build along with the others.

Give it a try:
```
cd tests
cabal configure
cabal test
```
then tweak `cabal.project` and try again...